lambdas/s3hash/tests/test_compute_checksum.py

```diff
@@ -140,7 +113,148 @@ async def test_legacy(s3_stub: Stubber):
         LOC.boto_args,
     )
 
+    mocker.patch("t4_lambda_s3hash.MULTIPART_CHECKSUMS", False)
+
     res = await s3hash.compute_checksum(LOC)
 
     checksum_hex = bytes.fromhex("d9d865cc54ec60678f1b119084ad79ae7f9357d1c4519c6457de3314b7fbba8a")
     assert res == s3hash.ChecksumResult(checksum=s3hash.Checksum.singlepart(checksum_hex))
+
+
+async def test_mpu_fail(s3_stub: Stubber):
+    ETAG = "test-etag"
+    SIZE = 1048576
+    s3_stub.add_response(
+        "get_object_attributes",
+        {"ObjectSize": SIZE, "ETag": ETAG},
+        EXPECTED_GETATTR_PARAMS,
+    )
+
+    s3_stub.add_client_error(
+        "create_multipart_upload",
+        "TestError",
+        expected_params=EXPECTED_MPU_PARAMS,
+    )
+
+    error = None
+    res = None
+    try:
+        res = await s3hash.compute_checksum(LOC)
+    except s3hash.LambdaError as e:
+        error = e
```

Comment:
I think you could use [`pytest.raises()`](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions).